met-api/src/met_api/services/engagement_content_service.py

```diff
@@ -0,0 +1,136 @@
+"""Service for engagement content management."""
+from http import HTTPStatus
+
+from met_api.constants.engagement_content_type import EngagementContentType
+from met_api.constants.membership_type import MembershipType
+from met_api.exceptions.business_exception import BusinessException
+from met_api.models.engagement_content import EngagementContent as EngagementContentModel
+from met_api.schemas.engagement_content import EngagementContentSchema
+from met_api.services import authorization
+from met_api.utils.roles import Role
+
+
+class EngagementContentService:
+    """Engagement content management service."""
+
+    @staticmethod
+    def get_contents_by_engagement_id(engagement_id):
+        """Get contents by engagement id."""
+        engagement_content_schema = EngagementContentSchema(many=True)
+        engagement_content_records = EngagementContentModel.get_contents_by_engagement_id(engagement_id)
+        engagement_contents = engagement_content_schema.dump(engagement_content_records)
+        return engagement_contents
+
+    @staticmethod
+    def create_engagement_content(engagement_content_data, engagement_id):
+        """Create engagement content item."""
+        one_of_roles = (
+            MembershipType.TEAM_MEMBER.name,
+            Role.EDIT_ENGAGEMENT.value
+        )
+        authorization.check_auth(one_of_roles=one_of_roles, engagement_id=engagement_id)
+
+        if engagement_content_data.get('engagement_id', None) != int(engagement_id):
+            raise ValueError('engagement content data has engagement id for a different engagement')
+
+        sort_index = EngagementContentService._find_higest_sort_index(engagement_id)
+
+        engagement_content_data['sort_index'] = sort_index + 1
+        created_content = EngagementContentService._create_content(engagement_id, engagement_content_data)
+        created_content.commit()
+        return EngagementContentSchema().dump(created_content)
+
+    @staticmethod
+    def _find_higest_sort_index(engagement_id):
```

Comment:
In other words, does this method find the "last" engagement content with regards to sort order?

Reply:
Yes